marklogic-data-hub/src/main/java/com/marklogic/hub/impl/ArtifactManagerImpl.java

```diff
@@ -70,16 +76,28 @@ public ObjectNode getArtifactSettings(String artifactType, String artifactName)
     }
 
     public ObjectNode updateArtifactSettings(String artifactType, String artifactName, JsonNode settings) {
-        return (ObjectNode) getArtifactService().setArtifactSettings(artifactType, artifactName, settings);
+        ObjectNode resp = (ObjectNode) getArtifactService().setArtifactSettings(artifactType, artifactName, settings);
+        writeArtifactInProject(artifactType, (ObjectNode) settings, true);
+        return resp;
     }
 
     protected ArtifactService getArtifactService() {
         DatabaseClient dataServicesClient = hubConfig.newStagingClient(null);
         return ArtifactService.on(dataServicesClient);
     }
 
-    protected void writeArtifactInProject(String artifactType, ObjectNode artifact) {
-        Path fileLocation = buildArtifactProjectLocation(artifactType, getNameFromArtifact(artifactType, artifact), getVersionFromArtifact(artifactType, artifact));
+    protected void writeArtifactInProject(String artifactType, ObjectNode artifact, boolean isSetting) {
+        Path fileLocation;
+        if (!isSetting) {
+            fileLocation = buildArtifactProjectLocation(artifactType, getNameFromArtifact(artifactType, artifact), getVersionFromArtifact(artifactType, artifact), false);
+        } else {
+            if (artifact.get("artifactName") == null ||  artifact.get("artifactName").asText().length() == 0) {
```

Comment:
Use !artifact.has("artifactName") . 

Also - would this really ever happen? And if it did, wouldn't we want to throw an error? Just logging it and moving on doesn't seem good, as something is really wrong if an artifact doesn't have a name. 

I think this code can assume that artifactName exists. If it doesn't, either the proceeding code will throw an error, or we can eagerly throw an exception. But it seems like the call to save the artifact in ML would have already failed too, right?

Reply:
yes, it is why i just log it and stop here without throwing an exception. I can change !artifact.has("artifactName")

Reply:
I'm thinking we do want an exception thrown though. It's a big deal if an artifact can't be written to the filesystem, right? A remote user will never see that log message.

Reply:
@rjrudin it is a artifact setting info and should always have an artifactName associated. We double-check it (may be bad data imported) because we do not want to store settings without any artifact. I can throw an exception and I am not sure what exception we should throw.